code/modules/mod/modules/module_pathfinder.dm

```diff
@@ -21,7 +21,13 @@
 	implant = new(src)
 
 /obj/item/mod/module/pathfinder/Destroy()
-	implant = null
+	QDEL_NULL(implant)
+	return ..()
+
+/obj/item/mod/module/pathfinder/Exited(atom/movable/gone, direction)
+	if(gone == implant)
+		implant = null
+		icon_state = "pathfinder_empty"
```

Comment:
Should put the icon state part in `update_icon_state`

Reply:
smart thinking smellbert